### PR TITLE
Based on #15983: Stores hash of pod manifest in mirror pod's annotation for release 1.0

### PR DESCRIPTION
**Pull request by dchen1107**
cc/ @yujuhong

### CODE REVIEW

**Comment by dchen1107** [comment]
Forgot including newly created pod_update.go earlier. Done!

**Comment by yujuhong** [comment]
> Forgot including newly created pod_update.go earlier. Done!

Hmm...instead of including this file, you should simply add `ConfigHashAnnotationKey` to pkg/kubelet/types.go because most variables have been already defined there. There are a few places in pkg/kubelet/config that you'd need to change `kubetypes.` to `kubelet.` because of this.

**Comment by ihmccreery** [comment]
Kicking Travis.

**Comment by dchen1107** [comment]
This is for 1.0 release. cc/ @thockin @bgrant0607

File: pkg/kubelet/config/file_test.go [hunk 1]
```diff
@@ -92,10 +93,11 @@ func TestReadPodsFromFile(t *testing.T) {
...
 			expected: CreatePodUpdate(kubelet.SET, kubelet.FileSource, &api.Pod{
 				ObjectMeta: api.ObjectMeta{
-					Name:      "test-" + hostname,
-					UID:       "12345",
-					Namespace: "mynamespace",
-					SelfLink:  getSelfLink("test-"+hostname, "mynamespace"),
+					Name:        "test-" + hostname,
+					UID:         "12345",
+					Namespace:   "mynamespace",
+					Annotations: map[string]string{kubetypes.ConfigHashAnnotationKey: "12345"},
```
Comment (yujuhong) [inline pkg/kubelet/config/file_test.go]: Unless I am mistaken, s/kubetypes/kubelet

File: pkg/kubelet/config/http_test.go [hunk 2]
```diff
@@ -149,11 +150,11 @@ func TestExtractPodsFromHTTP(t *testing.T) {
...
 					ObjectMeta: api.ObjectMeta{
-						UID:       "111",
-						Name:      "foo" + "-" + hostname,
-						Namespace: "mynamespace",
-
-						SelfLink: getSelfLink("foo-"+hostname, "mynamespace"),
+						UID:         "111",
+						Name:        "foo" + "-" + hostname,
+						Namespace:   "mynamespace",
+						Annotations: map[string]string{kubetypes.ConfigHashAnnotationKey: "111"},
```
Comment (yujuhong) [inline pkg/kubelet/config/http_test.go]: Same here. s/kubetypes/kubelet and remove the kubetypes import.

File: pkg/kubelet/config/file_test.go (hunk 1 above)
Comment (dchen1107) [inline pkg/kubelet/config/file_test.go]: Yeah. So messy here and local run didn't complain. Weird.

Comment (yujuhong) [inline pkg/kubelet/config/file_test.go]: You probably still had the old pod_udpates.go file in your local directory :)

**Comment by yujuhong** [comment]
lgtm. Thanks for handling the tricky cherry pick!

**Comment by ihmccreery** [comment]
Bypassing Shippable.  @jlowdermilk will run unit & integration tests manually.

**Pull request merged**